screens/profilePage.js

```diff
@@ -1,26 +1,199 @@
 import React, { Component } from 'react';
-import { View, StyleSheet } from 'react-native';
-import {Button} from 'react-native-paper';
+import { View, StyleSheet, SafeAreaView, ScrollView } from 'react-native';
 import { firebase }  from '../utils/firebase';
+import { Avatar,Provider, TextInput, RadioButton,Text, Subheading,Title, Card, Button,Paragraph, Dialog, Portal } from 'react-native-paper';
 
 class profilePage extends Component{
   // TODO: add more user info later once the design has finalized
   constructor(props) {
     super(props);
+    this.state = {
+      userInfo: {},
+      userId: firebase.auth().currentUser.uid,
+      edit: false,
+      new_name: null,
+      new_major: null,
+      new_year: null
+    }
+    this.get_profile_comp = this.get_profile_comp.bind(this);
+    this.edit = this.edit.bind(this)
+    this.save = this.save.bind(this)
+    this.cancel_edit = this.cancel_edit.bind(this)
   }
 
+  edit = () => {
+    this.setState({edit: true});
+  }
+  cancel_edit = () => {
+    this.setState({edit: false});
+  }
+
+  save = () => {
+    var new_userInfo = this.state.userInfo
+    const userId = this.state.userId
+    this.setState({edit: false})
+    const db = firebase.database().ref('/users')
+    if (this.state.new_name !== null) {
+      const new_name = this.state.new_name.split(' ')
+      const first = new_name[0]
+      const last = new_name[1]
+      db.child('/'+userId +'/first_name').set(first)
+      db.child('/'+userId +'/last_name').set(last)
+      new_userInfo.first_name = first
+      new_userInfo.last_name = last
+    }
+    if (this.state.new_major !== new_userInfo.major) {
+      const new_major = this.state.new_major
+      db.child('/'+userId +'/major').set(new_major)
+      new_userInfo.major = new_major
+    }
+    if (this.state.new_year !== new_userInfo.year) {
+      const new_year = this.state.new_year
+      db.child('/'+userId +'/year').set(new_year)
+      new_userInfo.year = new_year
+    }
+    this.setState({userInfo: new_userInfo})
+
+
+  }
+
+
   logout = () => {
     firebase.auth().signOut()
     this.props.navigation.replace('loginPage');
   }
 
+  componentDidMount() {
+    const userId = this.state.userId
+    console.log('userID: ',userId)
+    const db = firebase.database().ref('/users/'+userId);
+    var user = {}
+    db.on('value', (snapshot) => {
+      if (snapshot.exists()) {
+        user = snapshot.toJSON()
+        console.log('user: ',user)
+      }
+      this.setState({userInfo: user,
+                    new_year: user.year,
+                  new_major: user.major,
+                  new_name: user.first_name+' '+user.last_name
+                  })
+      console.log('userInfo: ',this.state.userInfo)
+      });
+  };
+
+  get_profile_comp = () => {
+    const user = firebase.auth().currentUser
+    console.log('user: ',user)
+    if (user.photoURL === '..') {
```

Comment:
not sure why are we making this comparison? Does user.photoURL ever have ".."?

Reply:
For now photoURL isn't generating an image so until I figure that out I left that there so it automatically makes the initial icon